knimin/lib/vioscreen.py

```diff
@@ -0,0 +1,446 @@
+import requests
+import os
+import functools
+import json
+import pandas as pd
+
+from datetime import datetime
+from knimin import config
+from knimin.lib.data_access import SQLHandler
+
+class VioscreenHandler(object):
+    def __init__(self):
+        with open('./auth.json') as f:
+            self._key = json.load(f)['key']
+            self._user = json.load(f)['user']
+            self._pw = json.load(f)['pw']
+        self._session = requests.Session()
+        # setup our HTTP header data
+        self._headers = None
+        self._users = None
+        self.sql_handler = SQLHandler(config)
+
+    # get an API token
+    def get_token(self):
+        """Gets an API token for vioscreen
+
+        Return
+        ------
+        str
+            The API token
+        """
+        return self.post('https://api.viocare.com/%s/auth/login' % self._key,
+                     data={"username": self._user, 
+                           "password": self._pw})['token']
+    
+    def get_users(self):
+        """Gets list of users that vioscreen has data for
+
+        Return
+        ------
+        list of dict
+            List of users that have vioscreen data
+        """
+        if not self._headers:
+            self._headers = {'Accept': 'application/json',
+                             'Authorization': 'Bearer %s' % self.get_token()}
+        return self.get('https://api.viocare.com/%s/users' % self._key,
+                         headers=self._headers)
+
+    def get(self, url, retries=5, **kwargs):
+        for i in range(retries):
+            req = self._session.get(url, **kwargs)
+            if req.status_code != 200:  # HTTP status code, 200 is all good
+                data = req.json()
+
+                # if we did not get a HTTP status code 200, than guess that the
+                # API token is no longer valid so get a new one and retry
+                if 'Code' in data and data['Code'] == 1016:
+                    self._headers['token'] = self.get_token()
```

Comment:
It looks like it's possible for `self._headers` to be `None`. For example:

```python
obj = VioscreenHandler()
foo = obj.get(...)
```

...in which case, the code would probably through a `KeyError`?